python/create_service_diagnostics_bundle.sh

```diff
@@ -74,43 +76,37 @@ function container_run () {
          ${CONTAINER_DCOS_SERVICE_DIAGNOSTIC_VOLUME_MOUNT} \
          -e PYTHONPATH=${CONTAINER_DCOS_SERVICE_DIAGNOSTICS_DIRECTORY}/dcos-commons/testing:${CONTAINER_DCOS_SERVICE_DIAGNOSTICS_DIRECTORY} \
          "${DOCKER_IMAGE}" \
-         sh -l -c "${command}"
+         sh -l -c "${*:-}"
 }
 
-function usage () {
-  container_run "${CONTAINER_SCRIPT_PATH} --help"
+function show_version() {
+  # Print script version
+  echo "${VERSION}"
 }
 
-if [ "${#}" -eq 1 ] && [[ "${1}" =~ ^(--help|-help|help|--h|-h)$ ]]; then
-  usage
-  exit 0
-fi
+function show_usage() {
+  # Show help for create_service_diagnostics_bundle.py
+  run_container "./${PYTHON_SCRIPT_NAME} --help"
+}
 
-echo "Initializing diagnostics..."
-
-DCOS_CLUSTER_MAJOR_MINOR_VERSION=$(dcos --version | awk -F"=" '/^dcos.version=/ {split($2,verPart,".");print verPart[1]"."verPart[2]}')
-echo "Detected attached DC/OS cluster major and minor version as '${DCOS_CLUSTER_MAJOR_MINOR_VERSION}'"
-
-readonly SUPPORTED_DCOS_VERSIONS="
-1.10
-1.11
-1.12
-1.13
-1.14
-2.0"
-if ! echo "${SUPPORTED_DCOS_VERSIONS}" | grep -qx "${DCOS_CLUSTER_MAJOR_MINOR_VERSION}"; then
-  echo "DC/OS ${DCOS_CLUSTER_MAJOR_MINOR_VERSION}.x is not supported by this tool."
-  echo "Supported DC/OS versions: ${SUPPORTED_DCOS_VERSIONS}"
-  exit 1
+# ###
+# 4. section: Main script execution
+# ###
+if [ "${#}" -eq 1 ]; then
+    case "$1" in
+        --version|-version|version|--v|-v)
+            show_version
+            exit 0
+            ;;
+        --help|-help|help|--h|-h)
+            show_usage
+            exit 0
+            ;;
+    esac
 fi
 
-container_run "rm -rf ${CONTAINER_DCOS_CLI_DIRECTORY} && mkdir ${CONTAINER_DCOS_CLI_DIRECTORY}
-               cd ${CONTAINER_DCOS_CLI_DIRECTORY_RO}
-               find . \
-                 \( -name 'dcos.toml' -or -name 'attached' \) \
-                 -exec cp --parents \{\} ${CONTAINER_DCOS_CLI_DIRECTORY} \;
-               cd /
-               dcos plugin add /tmp/dcos-core-cli-${DCOS_CLUSTER_MAJOR_MINOR_VERSION}.zip
-               ${CONTAINER_SCRIPT_PATH} ${*} \
-               --bundles-directory ${CONTAINER_BUNDLES_DIRECTORY} \
+mkdir -p "${BUNDLES_DIRECTORY}"
+
+run_container "./${PYTHON_SCRIPT_NAME} ${*} \
```

Comment:
IMO this function was better named as `container_run`. I'd expect a function named `run_container` to take a container name or id as argument. `container_run` indicates, to me, that the arguments are a script to be run. i.e. "container_run script.py".

Reply:
Makes sense. `run_in_container`?

Reply:
done